zebra-state/src/service/finalized_state.rs

```diff
@@ -73,6 +73,10 @@ impl FinalizedState {
             debug_stop_at_height: config.debug_stop_at_height.map(block::Height),
         };
 
+        // Make sure the database returned from the `db_config()` is always the same
+        // as the one returned by the `path()` method of the FinalizedState.
+        assert_eq!(new_state.path(), path);
```

Comment:
I added this just to avoid the dead code warning and i think it will not hurt but i can remove it if it is a redundant or incorrect check.

Reply:
Currently this happens to be true:
https://docs.rs/rocksdb/0.16.0/src/rocksdb/db.rs.html#434

But I don't think Zebra should enforce having exactly the same path. For example, it would be ok for the database to simplify the path.

Just use `#[allow(dead_code)]` on the `path` method instead.

Reply:
Ok, thank you. https://github.com/ZcashFoundation/zebra/pull/2349/commits/ea845cf798d199523b1a7b599f1e0b6a3a0f3f76